docs/docs/faq/index.md

```diff
@@ -10,4 +10,53 @@ version. Fetching an aggregate in a specific version directly from the `Aggregat
 first, but it is actually an anti-pattern in event sourcing. Basically what you're doing is a read action. This is why
 it does not belong to `AggregateRootRepository` but should be accomplished with
 [a projection or a read model](/docs/reacting-to-events/projections-and-read-models/). A read model does not have to be
-persistent, you can totally create an in-memory read model depending on your requirements.
\ No newline at end of file
+persistent, you can totally create an in-memory read model depending on your requirements.
+
+> How do I validate an aggregate against a service?
+
+As the aggregate is tasked with "maintaining the integrity of the model", you might be tempted to try to implement
+validation against an external service. Let's say we had an aggregate representing a bank account and wanted to
+validate the provided IBAN against an external API:
+
+```php
+<?php
+
+namespace AcmeCompany\AcmeProject;
+
+use EventSauce\EventSourcing\AggregateRoot;
+use EventSauce\EventSourcing\AggregateRootBehaviour;
+
+class AcmeBankAccount implements AggregateRoot
+{
+    use AggregateRootBehaviour;
+    
+    private string $iban;
+    
+    public function changeIban(string $iban): self
+    {
+        $externalApi->validateIban($iban); // TODO: FIXME
+        
+        $this->recordThat(new IbanWasChanged($iban));
+    }
+}
+```
+
+---
+TO BE ANSWERED:
+
+What do we do here now? There is no way to bring `$externalApi` to the aggregate. In my understanding, validation
+should thus probably happen before adding the IBAN to the aggregate. However, that kind of contradicts the
+statement of "the aggregate's responsibility is to maintain the integrity of the model"?
+That's why I think it deserves an FAQ entry :-)
```

Comment:
I think it should probably be encapsulated in a command but the docs lack a bit of information on that. Or I cannot find it ;)

Reply:
So would it be something along the lines of

```php
class ChangeIbanCommand
{
    public function __construct(private string $iban, private IbanValidationApi $ibanValidationApi) {}
    
    /* @throws SomeException */
    public function validateIban(): void
    {
         $this->ibanValidationApi->validate($this->getIban());
    }

    public function getIban(): string
    {
        return $this->iban;
    }
}
```

And then

```php
class AcmeBankAccount implements AggregateRoot
{
    use AggregateRootBehaviour;
    
    public function changeIban(ChangeIbanCommand $command): self
    {
        try {
            $command->validateIban();
            $this->recordThat(new IbanWasChanged($command->getIban()));
        } catch (SomeException) {
            $this->recordThat(new ProvidedIbanWasInvalid($command->getIban()));
        }
    }
}
```

Reply:
My recommended approach is to pass the service into the aggregate. So:

```php
class AcmeBankAccount implements AggregateRoot
{
    use AggregateRootBehaviour;
    
    public function changeIban(ChangeIbanCommand $command, IbanValidation $ibanValidation): self
    {
        $iban = $command->iban();

        if ( ! $ibanValidation->ibanIsValid($iban)) {
            $this->recordThat(new ProvidedIbanWasInvalid($iban));
        } else {
            $this->recordThat(new IbanWasChanged($iban));
        }
    }
}
```

Commands are vehicles for information, they are more like events than they are like services. Their sole purpose is to represent the instruction or request from the system, so no business logic validation.

Reply:
I see, thanks for the feedback! I'll update the PR then :) Any other feedback is super welcome :)

Reply:
Updated.